### PR TITLE
Text search nav

### PR DESCRIPTION
**Pull request by Dananji**
Related issue: #513 

This PR adds the ability to see focused search hits using results navigation when there are multiple hits within a focused cue.

Example of search results navigation when the transcript is one block of text:
Screenshot 2024-07-09 at 5 11 46 PM

### CODE REVIEW

File: src/services/utility-helpers.js [hunk 1]
```diff
@@ -515,6 +507,15 @@ export function getLabelValue(label) {
...
 
+export function decodeHTML(labelText) {
+  return labelText
+    .replace(/&amp;/g, '&')
+    .replace(/&lt;/g, '<')
+    .replace(/&gt;/g, '>')
+    .replace(/&quot;/g, '"')
+    .replace(/&apos;/g, "'");
+};
+
```
Comment (cjcolvar) [inline src/services/utility-helpers.js:518]: It looks like we already have libraries (`entities`, `html-entities`, `he`, and `parse-entities`) installed that do html entity encoding/decoding.  What about replacing this with one of those or is a more limited decoding preferable?

Comment (Dananji) [reply src/services/utility-helpers.js:518]: Replacing this with one of these sounds good to me.

File: src/services/search.js [hunk 2]
```diff
@@ -343,5 +343,11 @@ export const useFocusedMatch = ({ searchResults }) => {
     }
   }, [searchResults.matchingIds, focusedMatchIndex]);
 
+  useEffect(() => {
+    if (searchResults.matchingIds.length && focusedMatchIndex > 0) {
+      setFocusedMatchIndex(null);
```
[the remark this answers is not in the document]
Comment (Dananji) [reply src/services/search.js:348]: Yes, I see now that `null` is used to indicate disabled state. You are correct, setting it to -1 seems like the best solution. Good catch!

[the remark this answers is not in the document]
Comment (Dananji) [reply src/services/search.js:348]: Setting this to -1 reset the index to 0 when switching between transcripts in the same canvas after navigating through hits. e.g. when changing transcripts with the results nav saying `2 of x` set the index to zero for the new transcript (`0 of y`).
So I didn't change it. For some reason the `null` works in this.

File: src/services/utility-helpers.js (hunk 1 above)
Comment (Dananji) [reply src/services/utility-helpers.js]: Replace it with `decode` from `html-entities`.

**Review by cjcolvar — APPROVED** [review]
This looks good as far as I can tell.

**Pull request merged**